api/tests/integration/domain/usecases/copy-target-profile_test.js

```diff
@@ -0,0 +1,93 @@
+
+      await databaseBuilder.commit();
+
+      // when
+      const newlyCreatedTargetProfileId = await copyTargetProfile({
+        targetProfileId: originTargetProfile.id,
+        targetProfileRepository,
+      });
+
+      // then
+      const newlyCreatedTargetProfile = await knex('target-profiles')
+        .where({ id: newlyCreatedTargetProfileId })
+        .first();
+
+      expect(newlyCreatedTargetProfileId).not.to.equal(originTargetProfile.id);
+      expect(newlyCreatedTargetProfile.createdAt).not.to.equal(originTargetProfile.createdAt);
+
+      expect(newlyCreatedTargetProfile.name).to.equal('Copie_de_' + originTargetProfile.name);
+      expect(newlyCreatedTargetProfile.category).to.equal(originTargetProfile.category);
+      expect(newlyCreatedTargetProfile.areKnowledgeElementsResettable).to.equal(
+        originTargetProfile.areKnowledgeElementsResettable,
+      );
+      expect(newlyCreatedTargetProfile.comment).to.equal(originTargetProfile.comment);
+      expect(newlyCreatedTargetProfile.description).to.equal(originTargetProfile.description);
+      expect(newlyCreatedTargetProfile.imageUrl).to.equal(originTargetProfile.imageUrl);
+      expect(newlyCreatedTargetProfile.isPublic).to.equal(originTargetProfile.isPublic);
+      expect(newlyCreatedTargetProfile.isSimplifiedAccess).to.equal(originTargetProfile.isSimplifiedAccess);
+      expect(newlyCreatedTargetProfile.outdated).to.equal(originTargetProfile.outdated);
+      expect(newlyCreatedTargetProfile.ownerOrganizationId).to.equal(originTargetProfile.ownerOrganizationId);
+    });
+
+    it('should copy the tubes', async function () {
+      // given
+      const originTargetProfile = databaseBuilder.factory.buildTargetProfile({ migration_status: 'AUTO' });
+      const tubesData = [
+        { tubeId: 'recpP9Uaz1x6qq95e', level: 4 },
+        { tubeId: 'recYFCpGlmwQwAONl', level: 4 },
+        { tubeId: 'recoPAkxe29Ru3r3h', level: 3 },
+        { tubeId: 'recInPPTW79jkUbEY', level: 3 },
+        { tubeId: 'rec6Ic2FdcxSRYkdn', level: 2 },
+      ];
+
+      tubesData.map(({ tubeId, level }) => {
+        databaseBuilder.factory.buildTargetProfileTube({ targetProfileId: originTargetProfile.id, tubeId, level });
+      });
+
+      await databaseBuilder.commit();
+
+      // when
+      const newlyCreatedTargetProfileId = await copyTargetProfile({
+        targetProfileId: originTargetProfile.id,
+        targetProfileRepository,
+      });
+
+      // then
+      const newlyCreatedTargetProfile = await knex('target-profiles')
+        .where({ id: newlyCreatedTargetProfileId })
+        .first();
+
+      const newlyCreatedTargetProfileTubes = await knex('target-profile_tubes').where({
+        targetProfileId: newlyCreatedTargetProfile.id,
+      });
+
+      expect(newlyCreatedTargetProfileTubes.length).to.equal(tubesData.length);
+      newlyCreatedTargetProfileTubes.forEach((tube) => {
+        delete tube.id;
+        delete tube.targetProfileId;
+      });
```

Comment:
Pour que ce soit plus clair :  
```suggestion
newlyCreatedTargetProfileTubes.forEach((targetProfileTube) => {
        delete targetProfileTube.id;
        delete targetProfileTube.targetProfileId;
```